bubble.rb

```diff
@@ -0,0 +1,49 @@
+def bubble_sort(arr)
+  i = 0
+  j = 0
+  while i < arr.size
+    j = i + 1
+    while j <= arr.size - 1
+      if arr[i] > arr[j]
+        temp = arr[i]
+        arr[i] = arr[j]
+        arr[j] = temp
+      end
+      j += 1
+    end
+    i += 1
+  end
+end
+
+arr = [54, 14, 90, 2, 25]
+
+bubble_sort(arr)
+
+p arr
+
+arr1 = %w[hi hello hey]
+p arr1
+
+def bubble_sort_by(arr)
+  i = 0
+  j = 0
+  while i < arr.size
+    j = i + 1
+    while j < arr.size
+      result = yield(arr[i], arr[j])
+      if result.positive?
+        temp = arr[j]
+        arr[j] = arr[i]
+        arr[i] = temp
+      end
+      j += 1
+    end
+    i += 1
+  end
+end
```

Comment:
* [ ] bubble_sort_by method is implemented and working  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))

- Your method is returning `nil` instead of the sorted array.
```ruby
arr1 = %w[hi hello hey]

bubble = bubble_sort_by(arr1) do |num, num1|
     num.length - num1.length
end
puts bubble
>> nil
```

Reply:
Hello 👋, thanks for your review and pointing that errors, we fix it now.